src/vocutouts/workers.py

```diff
@@ -75,111 +86,87 @@ class TaskTransientError(Exception):
-    #
-    # This tells Butler to register data types every time, which as above
-    # should not be necessary, but they do have to be registered once.
-    data_query_terms = []
-    for key, value in data_id.items():
-        if isinstance(value, int):
-            data_query_terms.append(f"{key}={value}")
+    # Currently, only a single data ID and a single stencil are supported.
+    assert len(dataset_ids) == 1
+    assert len(stencils) == 1
+
+    # Convert the stencils to SkyStencils.
+    sky_stencils: List[SkyStencil] = []
+    for stencil_dict in stencils:
+        if stencil_dict["type"] == "circle":
+            center = SkyCoord(
+                stencil_dict["center"]["ra"] * u.degree,
+                stencil_dict["center"]["dec"] * u.degree,
+                frame="icrs",
+            )
+            radius = Angle(stencil_dict["radius"] * u.degree)
+            stencil = SkyCircle.from_astropy(center, radius)
+        elif stencil_dict["type"] == "polygon":
+            ras = [v[0] for v in stencil_dict["vertices"]]
+            decs = [v[1] for v in stencil_dict["vertices"]]
+            vertices = SkyCoord(ras * u.degree, decs * u.degree, frame="icrs")
+            stencil = SkyPolygon.from_astropy(vertices)
         else:
-            data_query_terms.append(f"{key}='{value}'")
-    data_query = " AND ".join(data_query_terms)
-    result = subprocess.run(
-        [
-            "pipetask",
-            "run",
-            "-j",
-            "1",
-            "-b",
-            repository,
-            "--register-dataset-types",
-            "-t",
-            "lsst.pipe.tasks.calexpCutout.CalexpCutoutTask",
-            "-d",
-            data_query,
-            "--output",
-            output_collection,
-            "-i",
-            f"{positions_collection},{collection}",
-        ],
-        stdout=subprocess.PIPE,
-        stderr=subprocess.STDOUT,
-    )
-    if result.returncode != 0:
-        msg = (
-            f"Error Cutout failed with status {result.returncode}\n"
-            f"{result.stdout.decode()}"
-        )
+            msg = f'UsageError Unknown stencil type {stencil_dict["type"]}'
+            raise TaskFatalError(msg)
+        sky_stencils.append(stencil)
+
+    # Perform the cutout.
+    try:
+        result = backend.process_uuid(sky_stencils[0], UUID(dataset_ids[0]))
+    except Exception as e:
+        msg = f"Error Cutout processing failed\n{type(e).__name__}: {str(e)}"
         raise TaskTransientError(msg)
 
-    # Return a pointer to the result Butler collection.  This must be a dict
-    # representation of a vocutouts.uws.models.JobResult.
+    # Return the result URL.  This must be a dict representation of a
+    # vocutouts.uws.models.JobResult.
+    result_url = result.geturl()
+    result_scheme = urlparse(result_url).scheme
+    if result_scheme != "s3":
+        msg = f"Error Backend returned URL with scheme {result_scheme}, not s3"
+        raise TaskTransientError(msg)
```

Comment:
I don't follow the transient part of that error?

Reply:
This is transient under the UWS definition of transient, meaning that if the user sends the exact same request again later it could succeed. (It's some sort of coding error on the backend and presumably would go away when we patched the software.)

It's sort of an annoying distinction because, for example, for a generic exception generating the cutout, it's not clear whether to make that transient or fatal because we don't really know whether it's a problem with the parameters or with the backend code. I erred on the side of making all backend errors transient.

Reply:
Oh, actually, that's right, UWS never defines what transient or fatal are. So that was the interpretation that I put on it, and maybe that's wrong....